src/pl/plpgsql/src/pl_exec.c

```diff
@@ -893,6 +902,7 @@ static void
 plpgsql_exec_error_callback(void *arg)
 {
 	PLpgSQL_execstate *estate = (PLpgSQL_execstate *) arg;
+	gp_enable_gpperfmon = orig_gp_enable_gpperfmon;
```

Comment:
Are you sure that this callback will be called without exceptions?

Reply:
Yes, the idea was to reproduce the logic from other parts of gpdb where we save `gp_enable_gpperfmon` to restore it later, but the main difference is that `PG_TRY` is used there and `plpgsql_exec_error_callback` is used in plpgsql_exec_function. Accodring to `plpgsql_exec_error_callback`'s comment, it's purpose is the same is `PG_TRY`'s, it should catch `ereport`s, so it seems to me rather safe to restore `gp_enable_gpperfmon` there

Reply:
As we discussed, we are going to reuse sql function logic with pgtry/pgcatch

Reply:
Removed `gp_enable_gpperfmon` assignment from `plpgsql_exec_error_callback` and moved to `plpgsql_call_handler` where various plpgsql functions are called

Reply:
Logic was moved to _SPI_execute_plan